crates/ruff_python_formatter/src/expression/mod.rs

```diff
@@ -155,6 +135,200 @@ impl FormatRule<Expr, PyFormatContext<'_>> for FormatExpr {
     }
 }
 
+/// The comments below are trailing on the addition, but it's also outside the
+/// parentheses
+/// ```python
+/// x = [
+///     # comment leading
+///     (1 + 2)  # comment trailing
+/// ]
+/// ```
+/// as opposed to
+/// ```python
+/// x = [(
+///     # comment leading
+///     1 + 2  # comment trailing
+/// )]
+/// ```
+/// , where the comments are inside the parentheses. That is also affects list
+/// formatting, where we want to avoid moving the comments after the comma inside
+/// the parentheses:
+/// ```python
+/// data = [
+///     (
+///         b"\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+///         b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+///     ),  # Point (0 0)
+/// ]
+/// ```
+/// We could mark those comments as trailing in list but it's easier to handle
+/// them here too.
+///
+/// So given
+/// ```python
+/// x = [
+///     # comment leading outer
+///     (
+///         # comment leading inner
+///         1 + 2 # comment trailing inner
+///     ) # comment trailing outer
+/// ]
+/// ```
+/// we want to keep the inner an outer comments outside the parentheses and the inner ones inside.
+/// This is independent of whether they are own line or end-of-line comments, though end-of-line
+/// comments can become own line comments when we discard nested parentheses.
+///
+/// Style decision: When there are multiple nested parentheses around an expression, we consider the
+/// outermost parentheses the relevant ones and discard the others.
+fn format_with_parentheses_comments(
+    expression: &Expr,
+    node_comments: &LeadingDanglingTrailingComments,
+    f: &mut PyFormatter,
+) -> FormatResult<()> {
+    // First part: Split the comments
+
+    // TODO(konstin): This is copied from `parenthesized_range`, except that we don't have the
```

Comment:
I'm still not sure what to do about this, otherwise this PR is ready

Reply:
I don't mind the duplication. We could implement a helper that returns an Iterator over the parentheses (opening and closing) and reuse it in `parenthesized_range` and here.